walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,90 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    
+    // stores conflicts of all required attendees
+    ArrayList<TimeRange> conflicts = new ArrayList<>();
+    // will store available times 
+    ArrayList<TimeRange> free = new ArrayList<>();
+
+    // find conflicts of required attendees
+    for (Event event : events) {
+      for (String attendee : request.getAttendees()) {
+        if (event.getAttendees().contains(attendee)) {
+          conflicts.add(event.getWhen());
+        }
+      }
+    }
+
+    Collections.sort(conflicts, TimeRange.ORDER_BY_START);
+
+    // handle edge cases 
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()){
+      return free;
+    }
+    int num_conflicts = conflicts.size(); 
+    if (num_conflicts == 0) {
+      free.add(TimeRange.WHOLE_DAY);
+      return free;
+    }
+
+    int start = 0; int end;
```

Comment:
nit: name these `current` and `next`. Next should always just be `start + 1` since you should only be evaluating adjacent events (see later comment).